util/grub-mkconfig_lib.in

```diff
@@ -293,10 +293,22 @@ version_test_gt ()
   return "$?"
 }
 
+version_delphix_latest ()
```

Comment:
IMO, it'd be more clear if we just used this function directly, rather than using this within `version_find_latest`.. e.g. since we don't intended to use the existing `version_find_latest`, it feels more complicated to keep that existing logic, even though it should never be used on a Delphix appliance.

What I'd suggest, is we rename this function to `version_find_delphix` (or something similar), so it's clear this is delphix specific functionality, and then use that function directly wherever `version_find_latest` is currently used.

Or perhaps just replace the current implementation of `version_find_latest` with this new function, so callers don't have to be updated (which may be important in the case of syncing with upstream, if new callers are added).

I'd rather `grub-mkconfig` fail outright (and hopefully we catch this in testing), than fallback to logic that we know doesn't work properly for a delphix appliance.

Reply:
> I'd rather grub-mkconfig fail outright (and hopefully we catch this in testing), than fallback to logic that we know doesn't work properly for a delphix appliance.

I was trying to think of corner cases during initial install where something might run `grub-mkconfig` before delphix-entire is installed, but I agree it's probably better to handle that explicitly and fail if the logic doesn't work rather than using the incorrect kernel.

Reply:
I think one possible case where this may not work correctly, is the case where we only build the delphix-kernel package(s) and install them, without rebuilding and re-installing a new delphix-entire package.

I don't think we'll take this approach for actual product features, but we may for internal tools and/or tests.. e.g. zfs-precommit does this (if I remember correctly), but only for the ZFS package(s).

If we chose to create some new kernel-specific tests that took this same approach that zfs-precommit does (i.e. install the package only, and not update delphix-entire), I have a feeling this approach would not work for that case.

Reply:
Note that `zfs-precommit` has been deprecated and we now have to use `ab-pre-push --zfs-tests`.

I don't think we will be able to just build a new delphix-kernel package targeting a new kernel version and install it without going through a full appliance-build run, given that there's quite a bit of packages that need to be rebuilt (kernel itself, zfs, connstat, delphix-kernel) and all those packages would need to be fetched individually, then added to an ad-hoc apt repo, then installed via `apt install delphix-kernel`). All this to say is that it would be quite hard to accidentally deploy a new version of the kernel, and I don't expect that to happen without appliance-build being involved and therefore delphix-entire being updated.